src/braket/experimental/autoqasm/api.py

```diff
@@ -72,6 +73,40 @@ def _function_with_params(f: Callable) -> Callable[[Any], aq_program.Program]:
         return _function_with_params
 
 
+def gate(*args, declaration_only: bool = False) -> Callable[[Any], None]:
+    """Decorator that converts a function into a callable gate definition.
+
+    Args:
+        declaration_only (bool): If declaration_only is True, the provided body of the
+            gate definition is ignored, and only the gate signature is declared so that it can be
+            called from elsewhere in the program. This may be used, for example, if the program
+            is being combined with an external list of gate definitions. Defaults to False.
```

Comment:
I'm still a little confused about the use case. If `ccnot` is defined elsewhere, why would someone define it a second time?

Reply:
The gate may be defined may be in an external OpenQASM file. The user wants to call the gate in an AutoQASM program, but they do not want to provide an implementation for it.

Reply:
Ah, so it wouldn't appear in the resulting program, but it would run successfully. Could we do something simple and tailored just for that? Like
```
@aq.function
def main():
    my_gate = aq.gate(num_qubits=1, args=["lambda"])
    my_gate(0, lambda=3.14)
```

Reply:
No, I don't think the Braket SDK has this concept yet. One advantage of doing this with a decorated function is that you get a named function with named parameters. Also you could, say, build a library of all of your supported/native gate declarations (if you are a quantum hardware provider) and provide it to your users.

However, I think that you and @yitchen-tim have convinced me that the use case here isn't really clear or immediate enough yet. I'll remove the `declaration_only` option from the PR for now.